MathLib/CMakeLists.txt

```diff
@@ -60,25 +63,34 @@ add_library(MathLib STATIC ${SOURCES})
 
 set_target_properties(MathLib PROPERTIES LINKER_LANGUAGE CXX)
 
-target_link_libraries(MathLib INTERFACE
-	GeoLib
-	logog
-)
+if (CVODE_FOUND)
+	target_link_libraries(MathLib INTERFACE
+		GeoLib
+		logog
+		sundials_cvode
+		sundials_nvecserial
+	)
+else()
```

Comment:
It would be nice if you could fix the CMakeLists.txt, since I'm pretty short on time, and you already know what's to be done.